binary_search_tree/binary_search_tree.py

```diff
@@ -12,22 +12,57 @@ def __init__(self, value):
 
     # Insert the given value into the tree
     def insert(self, value):
-        pass
+        # self.left and/or self.right need to be valid nodes
+        # for us to call insert on them
+        if value < self.value:
+            # check if self.left is a valid node
+            if self.left:
+                self.left.insert(value)
+            else:
+                # we have found a valid parking spot
+                self.left = BinarySearchTree(value)
+        # otherwise, value >= self.value:
+        else:
+            if self.right:
+                self.right.insert(value)
+            else:
+                self.right = BinarySearchTree(value)
 
     # Return True if the tree contains the value
     # False if it does not
     def contains(self, target):
-        pass
+        if self.value == target:
+            return True
+        elif target < self.value and self.left is not None:
```

Comment:
Nice job avoiding nested conditionals. This kind of syntax is more explicit and readable imo.